src/doc/src/reference/specifying-dependencies.md

```diff
@@ -183,53 +184,95 @@ you need to specify is the location of the repository with the `git` key:
 regex = { git = "https://github.com/rust-lang/regex.git" }
 ```
 
-Cargo will fetch the `git` repository at this location then look for a
-`Cargo.toml` for the requested crate anywhere inside the `git` repository
-(not necessarily at the root --- for example, specifying a member crate name
-of a workspace and setting `git` to the repository containing the workspace).
+Cargo fetches the `git` repository at that location, then looks for
+`Cargo.toml` file for the requested crate anywhere inside the `git` repository.  
+For example, `cpp` and `cpp_common` are members of `rust-cpp` repo
+and can be referred to by the repo's root URL (`https://github.com/mystor/rust-cpp`).
+
+```toml
+cpp        = { git = "https://github.com/mystor/rust-cpp" }
+cpp_common = { git = "https://github.com/mystor/rust-cpp" }
+```
+
+The above rule does not apply to local paths specified via `path` attribute.
 
-Since we haven’t specified any other information, Cargo assumes that
-we intend to use the latest commit on the default branch to build
-our package, which may not necessarily be the main branch.
-You can combine the `git` key with the `rev`, `tag`, or `branch` keys to
-specify something else. Here's an example of specifying that you want to use
+#### Choice of commit
+
+
+
+Cargo assumes that we intend to use the latest commit on the default branch to build
+our package if we only specify the repo URL, as in the examples above.
+
+You can combine the `git` key with the `rev`, `tag`, or `branch` keys to be more specific about
```

Comment:
The original text was written in quite a narrow column. I lengthened it a bit. Happy to reformat if there is a requirement to keep it narrow.